guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,604 @@
+x_val, y_val = create_split(val_idxs)
+x_train, y_train = create_split(train_idxs)
+
+PRE_TRAIN_STEPS_PER_EPOCH = tf.data.experimental.cardinality(train_ds) // BATCH_SIZE
+PRE_TRAIN_STEPS_PER_EPOCH = int(PRE_TRAIN_STEPS_PER_EPOCH.numpy())
+
+print(
+    tabulate(
+        [
+            ["train", tf.data.experimental.cardinality(train_ds), None],
+            ["val", x_val.shape, y_val.shape],
+            ["query", x_query.shape, y_query.shape],
+            ["index", x_index.shape, y_index.shape],
+            ["test", x_test.shape, y_test.shape],
+        ],
+        headers=["Examples", "Labels"],
+    )
+)
+
+"""
+## Augmentations
+Self-supervised networks require at least two augmented "views" of each example. This can be created using a DataSet and an augmentation function. The DataSet treats each example in the batch as its own class and then the augment function produces two separate views for each example.
+
+This means the resulting batch will yield tuples containing the two views, i.e.,
+Tuple[(BATCH_SIZE, 32, 32, 3), (BATCH_SIZE, 32, 32, 3)].
+
+TensorFlow Similarity provides several random augmentation functions, and here we combine augmenters from the simCLR module to replicate the augmentations used in simsiam.
+"""
+
+"""
+## Augmentation
+
+Now that we have all of our datasets produced, we can move on to dataset augmentation.
+Using KerasCV, it is trivial to construct an augmenter that performs as the one
+described in the original SimSiam paper.  Lets do that below.
+"""
+
+target_size = (96, 96)
+crop_area_factor = (0.08, 1)
+aspect_ratio_factor = (3 / 4, 4 / 3)
+grayscale_rate = 0.2
+color_jitter_rate = 0.8
+brightness_factor = 0.2
+contrast_factor = 0.8
+saturation_factor = (0.3, 0.7)
+hue_factor = 0.2
+
+augmenter = keras_cv.layers.Augmenter(
+    [
+        cv_layers.RandomFlip("horizontal"),
+        cv_layers.RandomCropAndResize(
+            target_size,
+            crop_area_factor=crop_area_factor,
+            aspect_ratio_factor=aspect_ratio_factor,
+        ),
+        cv_layers.MaybeApply(
+            cv_layers.Grayscale(output_channels=3), rate=grayscale_rate
+        ),
+        cv_layers.MaybeApply(
+            cv_layers.RandomColorJitter(
+                value_range=(0, 255),
+                brightness_factor=brightness_factor,
+                contrast_factor=contrast_factor,
+                saturation_factor=saturation_factor,
+                hue_factor=hue_factor,
+            ),
+            rate=color_jitter_rate,
+        ),
+    ],
+)
+
+"""
+Next, lets pass our images through this pipeline.
+Note that KerasCV supports batched augmentation, so batching before
+augmentation dramatically improves performance
+
+"""
+
+
+@tf.function()
+def process(img):
+    return augmenter(img), augmenter(img)
+
+
+train_ds = train_ds.repeat()
+train_ds = train_ds.shuffle(1024)
+train_ds = train_ds.batch(BATCH_SIZE)
+train_ds = train_ds.map(process, num_parallel_calls=tf.data.AUTOTUNE)
```

Comment:
Since `tf.data.AUTOTUNE` is being reused, do you want to assign it to a separate variable?